__tests__/app.test.js

```diff
@@ -174,3 +174,112 @@ describe("GET /api/articles/:article_id/comments", () => {
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Responds with new posted comment object", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "You should probably shorten this article",
+    };
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
+          comment_id: 19,
+          votes: 0,
+          created_at: expect.any(String),
+          author: "rogersop",
+          body: "You should probably shorten this article",
+          article_id: 2,
+        });
+      });
+  });
+
+  test("400: Responds with error message if given empty comment", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "",
+    };
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Please enter a comment before posting");
+      });
+  });
+
+  test("400: Responds with error message if given empty username", () => {
+    const newComment = {
+      username: "",
+      body: "hi!",
+    };
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Please enter a comment before posting");
+      });
+  });
+
+  test("400: Responds with error message if given empty object", () => {
+    const newComment = {};
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+
+  test("404: Responds with error message if user does not exist", () => {
+    const newComment = {
+      username: "johnregex",
+      body: "I love regex!",
+    };
+
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not found");
+      });
+  });
+
+  test("404: Responds with error message if article does not exist", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "You should probably shorten this article",
+    };
+    return request(app)
+      .post("/api/articles/99999999/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not found");
+      });
+  });
+
+  test("400: Responds with error message if given invalid input type for article_id", () => {
+    const newComment = {
+      username: "rogersop",
+      body: "You should probably shorten this article",
+    };
+    return request(app)
+      .post("/api/articles/dogs/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Bad request");
+      });
+  });
+});
```

Comment:
Testing is very thorough here including consideration of some edge cases. Tests are well written and easy to follow.